src/viewProviders/base.ts

```diff
@@ -295,7 +295,7 @@ export abstract class BaseViewProvider<
     // will never be called and the message won't update
     const plural = this.totalItemCount > 1 ? "s" : "";
     if (this.searchMatches.size > 0) {
-      this.treeView.message = `Showing ${this.searchMatches.size} of ${this.totalItemCount} result${plural} for "${search}"`;
+      this.treeView.message = `Showing ${this.searchMatches.size} of ${this.totalItemCount} statement${plural} for "${search}"`;
```

Comment:
We may need to approach this differently since this is coming from the `BaseViewProvider`, which will eventually be the parent class for the Resources/Topics/Schemas views (along with the current Flink statements view). Maybe each view has a string property (singular and plural?) that would be user-facing to describe what they would show when search functionality is used?

Reply:
+1 for BaseViewProvider demanding readonly `singularResource` and `pluralResource`strings to be defined.

Reply:
good points, updated to be dynamic with the readonly strings